beacon-chain/operations/blstoexec/pool.go

```diff
@@ -0,0 +1,135 @@
+package blstoexec
+
+import (
+	"math"
+	"sync"
+
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+)
+
+type listNode struct {
+	value *ethpb.SignedBLSToExecutionChange
+	prev  *listNode
+	next  *listNode
+}
+
+type changesList struct {
+	first *listNode
+	last  *listNode
+	len   int
+}
+
+// PoolManager maintains pending and seen BLS-to-execution-change objects.
+// This pool is used by proposers to insert BLS-to-execution-change objects into new blocks.
+type PoolManager interface {
+	PendingBLSToExecChanges() []*ethpb.SignedBLSToExecutionChange
+	BLSToExecChangesForInclusion() []*ethpb.SignedBLSToExecutionChange
```

Comment:
What's the difference between this and the method above? Pending and awaiting inclusion feel the same to me as a new reader of this code

Reply:
> // PendingBLSToExecChanges returns all objects from the pool.

> // BLSToExecChangesForInclusion returns objects that are ready for inclusion at the given slot.
// This method will not return more than the block enforced MaxBlsToExecutionChanges.